runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -420,11 +468,31 @@ end
 ---     Set to `true` to run a complete parse of the source (Note: Can be slow!)
 ---     Set to `false|nil` to only parse regions with empty ranges (typically
 ---     only the root tree without injections).
---- @return table<integer, TSTree>
-function LanguageTree:parse(range)
+--- @param on_parse fun(trees?: table<integer, TSTree>, err?: string)? Function invoked when parsing completes.
```

Comment:
Can we reverse the order of the arguments here to match UV and LSP callbacks. So `err, trees`